google/resource_compute_instance_group_manager.go

```diff
@@ -526,43 +446,50 @@ func resourceComputeInstanceGroupManagerUpdate(d *schema.ResourceData, meta inte
 		return err
 	}
 
-	d.Partial(true)
-
-	if _, ok := d.GetOk("rolling_update_policy"); d.Get("update_strategy") == "ROLLING_UPDATE" && !ok {
-		return fmt.Errorf("[rolling_update_policy] must be set when 'update_strategy' is set to 'ROLLING_UPDATE'")
+	updatedManager := &computeBeta.InstanceGroupManager{
+		Fingerprint: d.Get("fingerprint").(string),
 	}
+	var change bool
 
-	// If target_pools changes then update
 	if d.HasChange("target_pools") {
-		targetPools := convertStringSet(d.Get("target_pools").(*schema.Set))
+		updatedManager.TargetPools = convertStringSet(d.Get("target_pools").(*schema.Set))
+		change = true
+	}
 
-		// Build the parameter
-		setTargetPools := &computeBeta.InstanceGroupManagersSetTargetPoolsRequest{
-			Fingerprint: d.Get("fingerprint").(string),
-			TargetPools: targetPools,
+	if d.HasChange("auto_healing_policies") {
+		if v, ok := d.GetOk("auto_healing_policies"); ok {
+			updatedManager.AutoHealingPolicies = expandAutoHealingPolicies(v.([]interface{}))
+			change = true
 		}
+	}
 
-		op, err := config.clientComputeBeta.InstanceGroupManagers.SetTargetPools(
-			project, zone, d.Id(), setTargetPools).Do()
+	if d.HasChange("version") {
+		updatedManager.Versions = expandVersions(d.Get("version").([]interface{}))
+		change = true
+	}
+
+	if d.HasChange("update_policy") {
+		updatedManager.UpdatePolicy = expandUpdatePolicy(d.Get("update_policy").([]interface{}))
+		change = true
+	}
 
+	if change {
+		op, err := config.clientComputeBeta.InstanceGroupManagers.Patch(project, zone, d.Get("name").(string), updatedManager).Do()
 		if err != nil {
-			return fmt.Errorf("Error updating InstanceGroupManager: %s", err)
+			return fmt.Errorf("Error updating managed group instances: %s", err)
 		}
 
-		// Wait for the operation to complete
-		err = computeSharedOperationWait(config.clientCompute, op, project, "Updating InstanceGroupManager")
+		err = computeSharedOperationWait(config.clientCompute, op, project, "Updating managed group instances")
 		if err != nil {
 			return err
 		}
-
-		d.SetPartial("target_pools")
 	}
 
-	// If named_port changes then update:
+	// named ports can't be updated through PATCH
```

Comment:
I assume you know this because you tried it and it didn't work? I think it's also worth calling out that we update the named ports on the instance group directly rather than on the manager.

Reply:
Yeah, I originally had it in PATCH, and it told me they're immutable.

When you say "calling out", you mean in the documentation, a comment, or something else?

Reply:
just in a comment!